libcudacxx/include/cuda/std/__utility/pod_tuple.h

```diff
@@ -0,0 +1,541 @@
+    _T0 __t0;                                                                                                   \
+    _CCCL_PP_REPEAT(_SizeSub1, _CCCL_TUPLE_DEFINE_ELEMENT, 1)                                                   \
+                                                                                                                \
+    template <class _Fn, class _Self, class... _Us>                                                             \
+    _CCCL_TRIVIAL_API static auto __apply(_Fn&& __fn, _Self&& __self, _Us&&... __us) noexcept(                  \
+      __is_nothrow_callable_v<_Fn,                                                                              \
+                         _Us...                                                                                 \
+                         _CCCL_TUPLE_CVREF_TPARAM(0) _CCCL_PP_REPEAT(_SizeSub1, _CCCL_TUPLE_CVREF_TPARAM, 1)>)  \
+      -> __call_result_t<_Fn,                                                                                   \
+                         _Us...                                                                                 \
+                         _CCCL_TUPLE_CVREF_TPARAM(0) _CCCL_PP_REPEAT(_SizeSub1, _CCCL_TUPLE_CVREF_TPARAM, 1)>   \
+    {                                                                                                           \
+      return static_cast<_Fn&&>(__fn)(                                                                          \
+        static_cast<_Us&&>(__us)... _CCCL_TUPLE_ELEMENT(0) _CCCL_PP_REPEAT(_SizeSub1, _CCCL_TUPLE_ELEMENT, 1)); \
+    }                                                                                                           \
+                                                                                                                \
+    template <size_t _Idx>                                                                                      \
+    _CCCL_API static constexpr auto __get_mbr_ptr() noexcept                                                    \
+    {                                                                                                           \
+      using __result_t _CCCL_NODEBUG_ALIAS =                                                                    \
+        __type_index_c<_Idx, __c<&__tupl::__t0> _CCCL_PP_REPEAT(_SizeSub1, _CCCL_TUPLE_MBR_PTR, 1)>;            \
+      return __result_t::value;                                                                                 \
+    }                                                                                                           \
+  }
+
+_CCCL_DEFINE_TUPLE(0);
+_CCCL_DEFINE_TUPLE(1);
+_CCCL_DEFINE_TUPLE(2);
+_CCCL_DEFINE_TUPLE(3);
+_CCCL_DEFINE_TUPLE(4);
+_CCCL_DEFINE_TUPLE(5);
+_CCCL_DEFINE_TUPLE(6);
+_CCCL_DEFINE_TUPLE(7);
+*/
+
+template <class _T0>
+struct __tupl<index_sequence<0>, _T0>
+{
+  _T0 __t0;
```

Comment:
These specializations are dropping the EBCO, is that worth it here?

Reply:
yes it is. good catch.